pyhms/demes/single_pop_eas/multiwinner.py

```diff
@@ -0,0 +1,190 @@
+    """
+    return np.argsort(preferences)
+
+
+class MWPolicy(Protocol):
+    def __call__(self, preferences: np.ndarray, k: int) -> np.ndarray:
+        ...
+
+
+class SNTVPolicy(MWPolicy):
+    """
+    Under SNTV we pick the k candidates with the highest plurality scores
+    (i.e., k candidates ranked first most frequently).
+    """
+
+    def __call__(self, preferences: np.ndarray, k: int) -> np.ndarray:
+        best_candidate_indices = preferences[:, 0].flatten()
+        unique_candidate_indices, counts = np.unique(best_candidate_indices, return_counts=True)
+        return unique_candidate_indices[np.argsort(-counts)[:k]]
+
+
+class BordaPolicy(MWPolicy):
+    """
+    Under k-Borda we pick k candidates with the highest Borda scores.
+    Borda score is defined as follows (N is the total number of candidates,
+    i is the position of the candidate in the preference list).
+    .. math::
+
+    \beta(i) = N - i
+    """
+
+    def __call__(self, preferences: np.ndarray, k: int) -> np.ndarray:
+        N = preferences.shape[1]
+        positions_in_sorted_array = get_positions_in_preferences(preferences)
+        borda_scores = (N - positions_in_sorted_array).sum(axis=0)
+        return np.argsort(-borda_scores)[:k]
+
+
+class BlocPolicy(MWPolicy):
+    """
+    Under Bloc we pick the k candidates with the highest k-Approval scores.
+    """
+
+    def __call__(self, preferences: np.ndarray, k: int) -> np.ndarray:
+        top_k = preferences[:, :k].flatten()
+        unique_candidate_indices, counts = np.unique(top_k, return_counts=True)
+        return unique_candidate_indices[np.argsort(-counts)[:k]]
+
+
+class CCGreedyPolicy(MWPolicy):
+    """
+    Chamberlin-Courant greedy algorithm by Lu and Boutilier.
+    """
+
+    def __call__(self, preferences: np.ndarray, k: int) -> np.ndarray:
+        N = preferences.shape[1]
+        positions_in_sorted_array = get_positions_in_preferences(preferences)
+        all_borda_scores = N - positions_in_sorted_array
+        winner_indices: list[int] = []
+        candidate_indices = np.arange(N)
+        np.random.shuffle(candidate_indices)
+        for _ in range(k):
+            best_score = -np.inf
+            best_candidate_index = None
+            np.random.shuffle(candidate_indices)
+            for candidate_index in candidate_indices:
+                if candidate_index in winner_indices:
+                    continue
+                new_winner_indices = winner_indices + [candidate_index]
+                scores_to_consider = all_borda_scores[new_winner_indices, :]
+                total_score = np.sum(np.max(scores_to_consider, axis=1))
+                if total_score > best_score:
+                    best_score = total_score
+                    best_candidate_index = candidate_index
+            winner_indices.append(best_candidate_index)
+        return np.array(winner_indices)
```

Comment:
All 4 of the policies look to be implemented correctly according to the source article. The Monroe greedy algorithm by Skowron is missing, although I don't think that it is necessary currently as it seems to produce similar effects as CCGreedy